native/rust/archetype/src/main/resources/archetype-resources/pom.xml

```diff
@@ -0,0 +1,83 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+
+    The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+    (the "License"). You may not use this work except in compliance with the License, which is
+    available at www.apache.org/licenses/LICENSE-2.0
+
+    This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+    either express or implied, as more fully set forth in the License.
+
+    See the NOTICE file distributed with this work for information regarding copyright ownership.
+
+-->
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+  <parent>
+    <groupId>org.alluxio</groupId>
+    <artifactId>alluxio-native-rust</artifactId>
+    <version>${project.version}</version>
+  </parent>
+
+  <artifactId>\${artifactId}</artifactId>
```

Comment:
are the `/$` intentional in this file?

Reply:
yes indeed. there are two passes of string interpolation:

1. the first is by maven resource plugin, when copying the resource files from the source directory to the output directory for packaging. Things like `${project.version}` should be kept in sync with the whole project and should not be customized when a user generates a submodule.
2. the second is by the archetype plugin, when a user actually uses the archetype to generate a submodule. Things like `${artifactId}` are meant to be different for different modules, so must be customized.

The escaping is used to make sure templates for the second interpolation are not eagerly processed by the resource plugin.